docs/sources/reference/api/docker_remote_api_v1.19.md

```diff
@@ -1253,12 +1253,15 @@ or being killed.
 Query Parameters:
 
 -   **dockerfile** - path within the build context to the Dockerfile. This is 
-        ignored if `remote` is specified and points to an individual filename.
+        ignored if `remote` is specified and points to an external dockerfile.
 -   **t** – repository name (and optionally a tag) to be applied to
         the resulting image in case of success
--   **remote** – A Git repository URI or HTTP/HTTPS URI build source. If the 
-        URI specifies a filename, the file's contents are placed into a file 
-		called `Dockerfile`.
+-   **remote** – A Git repository URI or HTTP/HTTPS context URI. If the
+        URI points to a single text file, the file's contents are placed into
+	a file called `Dockerfile` and the image is built with that file as the
+	context. If the URI points to a compressed tarball, the file is fetched
+	and the contents therein used as the context for the build. If `dockerfile`
+	is also specified, there must be a file with the corresponding name inside `remote`.
```

Comment:
@thaJeztah when you pass an URL to a remote tarball from the client, the `-f` option is used by the daemon to locate the Dockerfile inside the tarball, for example:

> `docker build -f dev/Dockerfile http://server/context.tar.xz`

the builder will look for a dev/Dockerfile inside context.tar.xz; if the URL points to a Dockerfile instead, the -f option is ignored.